include/internal/libspdm_macro_check.h

```diff
@@ -41,4 +41,37 @@
 #undef LIBSPDM_ASYM_ALGO_SUPPORT
 #undef LIBSPDM_DHE_ALGO_SUPPORT
 
+/*ensure at least one FIPS approved algorithm is supported*/
+#define LIBSPDM_FIPS_SIGNATURE_ALGO \
+    ((LIBSPDM_RSA_SSA_SUPPORT) || (LIBSPDM_RSA_PSS_SUPPORT) || (LIBSPDM_ECDSA_SUPPORT))
+
+#define LIBSPDM_FIPS_DHE_ALGO ((LIBSPDM_FFDHE_SUPPORT) || (LIBSPDM_ECDHE_SUPPORT))
```

Comment:
Some (most) of these algorithms are conditional on the endpoint's capabilities. For example if an endpoint does not support asymmetric key exchange then `LIBSPDM_FIPS_DHE_ALGO` can be `0 / false`.

Reply:
Thanks. I have deleted the `LIBSPDM_FIPS_DHE_ALGO` check.

Reply:
If we follow the capability, we should remove `LIBSPDM_FIPS_AEAD_ALGO` as well, since it is not required in authentication only use case.

We should remove `LIBSPDM_FIPS_SIGNATURE_ALGO` as well, since it is not required in PSK use case.

Then we only need `LIBSPDM_FIPS_HASH_ALGO`.

However, that at least one bit set according to capability should be enforced by the logic in https://github.com/DMTF/libspdm/blob/main/include/internal/libspdm_macro_check.h.
I dont think it is related to FIPS.

Reply:
Here is my suggestion:
1) We should improve the algo check with capability in https://github.com/DMTF/libspdm/blob/main/include/internal/libspdm_macro_check.h. E.g LIBSPDM_HASH_ALGO_SUPPORT is missing, LIBSPDM_AEAD_ALGO_SUPPORT is missing.

2) We dont need LIBSPDM_FIPS_xxx_ALGO, because at least one bit rule is already applied.

This should be a generic algo check, not specific FIPS algo check.

Reply:
Thanks. I have added the `LIBSPDM_HASH_ALGO_SUPPORT` and `LIBSPDM_AEAD_ALGO_SUPPORT`.
But for `LIBSPDM_AEAD_ALGO_SUPPORT`, I don't know how to add the check.